src/block/Cake.php

```diff
@@ -99,8 +100,9 @@ public function getDropsForCompatibleTool(Item $item) : array{
 	}
 
 	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
-		if($player !== null){
-			return $player->consumeObject($this);
+		if($player !== null && $player->consumeObject($this)){
+			$this->position->getWorld()->addSound($this->position->add(0.5, 0.5, 0.5), new BurpSound());
```

Comment:
This doesn't make a lot of sense. It's not the cake that's burping, it's the player.

This should be using `Player->broadcastSound()` so that it correctly respects the `silent` flag.

Reply:
In vanilla the burp is reproduced from the center of the block, not from the eating entity.